src/inverted-index.js

```diff
@@ -1,41 +1,72 @@
+class Index {
 
+	constructor(){
+		this.invertedIndex = {};
+	}
+//pick data from url
+	readJson(url) {
+			return fetch(url).
+				then(function(response) {
+				return response.json()
+			}).//turn data into json
+				then(function(data) {
+					return data;
+				}).catch (function(error) {
+					return false;
+			});
+		}
 
-function readjson(){
-	return fetch("../jasmine/books.json").then(function(response){
-		return response.json();
-	})
-	.catch(function(err){
-		console.log('There has been a problem with your fetch operation: ' + err);
-		return false;
-	});
-}
+	createIndex(url){
+		var sentence = [];
+		var self = this;
 
-function indexCreate(){
-	var index=[];
-	return fetch('../jasmine/books.json').then(function(response) {
-		console.log('I am def');
-		return response.json();
-	}).catch(function(error) {
-		return 'There was an error'
-	});
-	// fetch("../jasmine/books.json").then(function(response){
-	// 	response.json().then(function(data){
-	// 	data.map(function(x, y){
-	// 		//console.log("Index: "+ y +", title: " + x.title);
-	// 		index.push("Index: "+ y +", title: " + x.title);
+		return this.readJson(url).then(function(response) {
+			var data = response;
+			//get the documents in the json object and store the content in an array
+			data.map(function(element, index) {
+				var content = [];
+				var titles = Object.keys(element);
+				var i=0;
 
-	// 	});
-	// 	console.log(index);
-	// 	return index;
-		
-	// });
-		
+				for (i = 0; i < titles.length; i++) {
+					content.push(element[titles[i]]);
+				}
+				//split the content into words and store each word in the invertedIndex
+				content.map(function(phrase, indexOfPhrase) {
+					sentence = phrase.toLowerCase().replace(/\W/g, ' ').replace(/\s+/g, ' ').trim();
```

Comment:
**Code Styling - Critical**
Since you have a long chain here, this can be styled as so:
```javascript
sentence = phrase.toLowerCase()
  .replace(/\W+/g, ' ')
  .trim();
```
I have combined the two regular expressions into one as they are both replacing their matches with ' '